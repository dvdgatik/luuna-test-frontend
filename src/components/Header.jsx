import React from 'react'
import {Link} from 'react-router-dom';
import styled from 'styled-components';

 
const StyleHeader = styled.header`
	background: #002355;
	text-align: center;
	color:#ffffff;
	padding: 10px 20px;
`;

const Nav = styled.nav`
`;

const Span = styled.span`
	display:block;
`;

const Anchor = styled(Link)`
	display:inline-block;
	color: #ffffff;
	padding: 0 10px;
	font-size: ${props=>props.fontSize || '13pt' };
	text-decoration: none;
	font-weight: bold;
	&:hover ${Anchor} {
    	color:#ffffff;
  	};
`;  



const Header = props => {
	return(
		/*<header>
			<nav className="navbar">
				<span className="navbar-brand mb-0 h1">
				<a className='cool-link' src="/">Disease Classifier</a>
				</span>
				<span>See individual: 
				<a className='cool-link' src="/hypertension">Hypertension</a>
				<a className='cool-link' src="/kydney" >Kidney Desease</a>
				</span>
			</nav>
		</header>*/
		<StyleHeader>
			<Nav className="navbar">
				<Span className="navbar-brand mb-0 mr-0 h1">
					<Anchor fontSize='24pt' to='/'>Luuna</Anchor>
				</Span>
				<Span>
					<Anchor  to='/users'>Users</Anchor>
					<Anchor  to='/repositories'>Repositories</Anchor>
				</Span>
			</Nav>
		</StyleHeader>
	);
}






export default Header;



