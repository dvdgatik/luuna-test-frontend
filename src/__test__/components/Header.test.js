import React from 'react';
import {mount, shallow} from 'enzyme';
import Header from '../../components/Header';
import styled from 'styled-components'
import {create} from 'react-test-renderer';
import 'jest-styled-components';
import { BrowserRouter as Router } from 'react-router-dom';
describe('<Header>',()=>{
	const header = shallow(<Router><Header/></Router>);
	test('Render of Header Component', () => {
		expect(header.length).toEqual(1);
	});
});

describe('Header Snapshot', () => {
	test('Check UI of Header Component',()=>{
		const header = create(<Router><Header/></Router>);
		expect(header.toJSON()).toMatchSnapshot();
	})
});