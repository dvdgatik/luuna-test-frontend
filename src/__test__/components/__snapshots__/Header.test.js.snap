// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot Check UI of Header Component 1`] = `
.c0 {
  background: #002355;
  text-align: center;
  color: #ffffff;
  padding: 10px 20px;
}

.c1 {
  display: block;
}

.c2 {
  display: inline-block;
  color: #ffffff;
  padding: 0 10px;
  font-size: 24pt;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: bold;
}

.c2:hover {
  color: #ffffff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2:after {
  content: '';
  display: block;
  width: 0;
  height: 2px;
  margin: 5px auto;
  background: #fff;
  -webkit-transition: width .3s;
  transition: width .3s;
}

.c2:hover::after {
  width: 100%;
  -webkit-transition: width .3s;
  transition: width .3s;
}

.c3 {
  display: inline-block;
  color: #ffffff;
  padding: 0 10px;
  font-size: 13pt;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: bold;
}

.c3:hover {
  color: #ffffff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3:after {
  content: '';
  display: block;
  width: 0;
  height: 2px;
  margin: 5px auto;
  background: #fff;
  -webkit-transition: width .3s;
  transition: width .3s;
}

.c3:hover::after {
  width: 100%;
  -webkit-transition: width .3s;
  transition: width .3s;
}

<header
  className="c0"
>
  <nav
    className="navbar"
  >
    <span
      className="c1 navbar-brand mb-0 mr-0 h1"
    >
      <a
        className="c2"
        fontSize="24pt"
        href="/"
        onClick={[Function]}
      >
        Luuna
      </a>
    </span>
    <span
      className="c1"
    >
      <a
        className="c3"
        href="/users"
        onClick={[Function]}
      >
        Users
      </a>
      <a
        className="c3"
        href="/repositories"
        onClick={[Function]}
      >
        Repositories
      </a>
    </span>
  </nav>
</header>
`;
